doc/man3/BIO_push.pod

```diff
@@ -8,22 +8,25 @@ BIO_push, BIO_pop, BIO_set_next - add and remove BIOs from a chain
 
  #include <openssl/bio.h>
 
- BIO *BIO_push(BIO *b, BIO *append);
+ BIO *BIO_push(BIO *b, BIO *next);
  BIO *BIO_pop(BIO *b);
  void BIO_set_next(BIO *b, BIO *next);
 
 =head1 DESCRIPTION
 
-The BIO_push() function appends the BIO B<append> to B<b>, it returns
-B<b>.
+BIO_push() function prepends the BIO I<b>, which may be a chain of BIOs,
+to I<next> and returns I<b>.  In other words, it pushes I<b> on I<next>.
+I<next> may be NULL, which means that nothing is appended to I<b>.
+If I<b> is NULL the function does nothing and returns I<next>.
```

Comment:
Not sure that this text makes things less confusing, as you're talking about prepending in one sentence and appending in the next.  I would stick to the "append" explanation throughout, that did seems pretty clear.  However, the added explanation of what happens with NULL is quite helpful, do keep that.

Reply:
Also, I just remembered that the explanation isn't entirely correct either way.  A simple comment that exists in the code clarifies:

https://github.com/openssl/openssl/blob/f87b4c4ea67393c9269663ed40a7ea3463cc59d3/crypto/bio/bio_lib.c#L631-L632

Reply:
I agree that it was still a bit confusing to talk about both appending and prepending,
so re-worded the remaining use of the word "appending".

I do believe it is better to talk about prepending (and not about appending) because this fits much better with the stack analogy, which is the origin of the function names `BIO_push` and `BIO_pop`.

Doing so, I made the text even more precise, now mentioning also the control operations.